include/fluent-bit/flb_error_reporter.h

```diff
@@ -0,0 +1,51 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2021 The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <monkey/mk_core/mk_list.h>
+#include <fluent-bit/flb_sds.h>
+
+#include <time.h>
+
+#define STATUS_MESSAGE_FILE_PATH_ENV "STATUS_MESSAGE_FILE_PATH"
+#define STATUS_MESSAGE_TTL_ENV "STATUS_MESSAGE_TTL"
+#define STATUS_MESSAGE_MAX_BYTE_LENGTH_ENV "STATUS_MESSAGE_MAX_BYTE_LENGTH"
+#define STATUS_MESSAGE_TTL_DEFAULT 20
+#define STATUS_MESSAGE_MAX_BYTE_LENGTH_DEFAULT 1024
+
+struct flb_error_reporter {
```

Comment:
So I sent an email to Eduardo about this feature to make sure he's alright with it in core. I think we should make it clear that this is code for AWS. So may be "flb_aws_error_reporter". Then we can also hide it behind a CMake flag which defaults to off so that no one will compile this code unless they want it. 

I think we should may be also move the code files to the `aws` directory in Fluent Bit.

This is not a generic error reporter. It's very specific to our needs.

Reply:
Moved error reporting part of logic into /src/aws/ folder.  and renamed to flb_aws_error_reporter in every place.